Auth

Danila-Peshev opened the pull request.

Danila-Peshev committed:
Add auth;
delete field pawword type User;
Correct client link.

Danila-Peshev committed:
Del output in console

Isidora25 commented on src/components/AuthContext.tsx:
```diff
@@ -27,28 +29,37 @@ const AuthContext = createContext<AuthContextType>(defaultAuthContext);
   const [user, setUser] = useState<User | null>(null);
   const [componentIsReady, setComponentIsReady] = useState(false);
+  const [loginMutation] = useMutation(LOGIN);
+  const {data, loading, error} = useQuery(GET_USER)
```
вынеси все это в касотмный хук и положи рядома с описанием квери. Тем самым ты вынесешь работу с апи на отдельный уровени

Isidora25 commented on src/components/AuthContext.tsx:
```diff
@@ -27,28 +29,37 @@ const AuthContext = createContext<AuthContextType>(defaultAuthContext);
+    localStorage.setItem("token", responseMutation.data.login.token)
+    const responseQuery = await getUserQuery()
+    if (!responseQuery.error && responseQuery.data) {
+      setUser({id: responseQuery.data.getUser.id, email: responseQuery.data.getUser.email});
```
не надо так усложнять, ты можешь передернуть квери с помощью refetch

Isidora25 commented on src/components/AuthContext.tsx:
```diff
@@ -27,28 +29,37 @@ const AuthContext = createContext<AuthContextType>(defaultAuthContext);
-      localStorage.setItem("user", JSON.stringify(optinalUser));
-      setUser(optinalUser);
+    const responseMutation = await loginMutation({variables: {email, password}})
+    localStorage.setItem("token", responseMutation.data.login.token)
```
проверь, тут в случае ошибки может не быть объекта data

Isidora25 commented on src/components/AuthContext.tsx:
```diff
@@ -27,28 +29,37 @@ const AuthContext = createContext<AuthContextType>(defaultAuthContext);
       return true;
     }
-    return false;
+    return false
```
это тебе уже не нужно скорее всего

Danila-Peshev committed:
Commit:
Moved query/mutation hooks to a custom auth hook;
Set up refetch and shortened the code.

Isidora25 commented on src/gql-hooks/useLogin.ts:
```diff
@@ -0,0 +1,26 @@
+import { LOGIN } from "../queries-mutations/mutations";
+import { GET_USER } from "../queries-mutations/queries";
+
+const useLogin = () => {
+  const [loginMutation] = useMutation(LOGIN);
+  const {
+    data: dataGetUser,
+    loading: isLoadingGetUser,
+    error: errorGetUser,
+    refetch: refetchGetUser,
+  } = useQuery(GET_USER, {
+    fetchPolicy: "no-cache",
+    notifyOnNetworkStatusChange: true,
+  });
+
+  return {
+    loginMutation,
+    dataGetUser,
+    isLoadingGetUser,
+    errorGetUser,
+    refetchGetUser,
+  };
+};
+
```
сделай два разных хука 
1. useLogin
2. useGetUser

если тебе придется вызывать их по отдельности, будет неудобно

Isidora25 commented on src/gql-hooks/useLogin.ts:
```diff
@@ -0,0 +1,26 @@
+    error: errorGetUser,
+    refetch: refetchGetUser,
+  } = useQuery(GET_USER, {
+    fetchPolicy: "no-cache",
```
cache-and-network не работает?
"no-cache" ты вырубаешь кеширование совсем, а значит вырубаешь оптимизацию

Danila-Peshev committed:
Split the auth hooks.
